### ISSUE TITLE
Readiness probe failed: Get "http://10.1.0.16:8123/health/ready": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

### ISSUE THREAD

**Issue by philthynz** [Azure/application-gateway-kubernetes-ingress#1052]
**Describe the bug**
ingress-azure has an error in the event logs:

```
Readiness probe failed: Get "http://10.1.0.16:8123/health/ready": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

**To Reproduce**
Use Terraform to deploy container:

```
resource "helm_release" "ingress-azure" {
  name       = "ingress-azure"
  repository = "https://appgwingress.blob.core.windows.net/ingress-azure-helm-package"
  chart      = "ingress-azure"
  timeout    = 120

  set {
  name  = "appgw.name"
  value = azurerm_application_gateway.app_gateway.name
  }

  set {
  name  = "appgw.resourceGroup"
  value = azurerm_kubernetes_cluster.aks.node_resource_group
  }

  set {
  name  = "appgw.subscriptionId"
  value = data.azurerm_subscription.current.subscription_id
  }

  set {
  name  = "appgw.usePrivateIP"
  value = false
  }

  set {
  name  = "appgw.shared"
  value = false
  }

  set {
  name  = "armAuth.type"
  value = "aadPodIdentity"
  }

  set {
  name  = "rbac.enabled"
  value = true
  }

  set {
  name  = "armAuth.identityResourceID"
  value = azurerm_kubernetes_cluster.aks.kubelet_identity[0].user_assigned_identity_id
  }

  set {
  name  = "armAuth.identityClientID"
  value = azurerm_kubernetes_cluster.aks.kubelet_identity[0].client_id
  }
}
```

The "aad-pod-identity" container has also been deployed with rbac:

```
 resource "helm_release" "aad-pod-identity" {
   name       = "aad-pod-identity"
   repository = "https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts"
   chart      = "aad-pod-identity"
   timeout    = 120

   set {
   name  = "rbac.enabled"
   value = "true"
   }
}
```

**Ingress Controller details**

```
Name:         ingress-azure-554dcb67d4-wvcz5
Namespace:    default
Priority:     0
Node:         aks-default-73418808-vmss000000/10.1.0.4
Start Time:   Thu, 05 Nov 2020 15:53:19 +1300
Labels:       aadpodidbinding=ingress-azure
              app=ingress-azure
              pod-template-hash=554dcb67d4
              release=ingress-azure
Annotations:  checksum/config: 10aad7f415f97ce65558e70062995f8c3a2bed8caa8d07aeba23f37f4eaf9105
              prometheus.io/port: 8123
              prometheus.io/scrape: true
Status:       Running
IP:           10.1.0.16
IPs:
  IP:           10.1.0.16
Controlled By:  ReplicaSet/ingress-azure-554dcb67d4
Containers:
  ingress-azure:
    Container ID:   docker://5ab4ed2f4893a12e41650ca9df3e2736925d85c090842822d8e9157769035a98
    Image:          mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.2.1
    Image ID:       docker-pullable://mcr.microsoft.com/azure-application-gateway/kubernetes-ingress@sha256:cf1df9645ded72127967c99bbf88b758b5b19b698ba18a9ed74c3a7ebf016ef1
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Thu, 05 Nov 2020 16:13:50 +1300
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Thu, 05 Nov 2020 16:03:36 +1300
      Finished:     Thu, 05 Nov 2020 16:13:48 +1300
    Ready:          False
    Restart Count:  2
    Liveness:       http-get http://:8123/health/alive delay=15s timeout=1s period=20s #success=1 #failure=3
    Readiness:      http-get http://:8123/health/ready delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment Variables from:
      ingress-azure  ConfigMap  Optional: false
    Environment:
      AZURE_CLOUD_PROVIDER_LOCATION:  /etc/appgw/azure.json
      AGIC_POD_NAME:                  ingress-azure-554dcb67d4-wvcz5 (v1:metadata.name)
      AGIC_POD_NAMESPACE:             default (v1:metadata.namespace)
    Mounts:
      /etc/appgw/azure.json from azure (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from ingress-azure-token-wdwxb (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  azure:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/kubernetes/azure.json
    HostPathType:  File
  ingress-azure-token-wdwxb:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ingress-azure-token-wdwxb
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                  From                                      Message
  ----     ------     ----                 ----                                      -------
  Normal   Scheduled  21m                  default-scheduler                         Successfully assigned default/ingress-azure-554dcb67d4-wvcz5 to aks-default-73418808-vmss000000
  Normal   Pulling    21m                  kubelet, aks-default-73418808-vmss000000  Pulling image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.2.1"
  Normal   Pulled     21m                  kubelet, aks-default-73418808-vmss000000  Successfully pulled image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.2.1" in 1.080920974s
  Normal   Created    21m                  kubelet, aks-default-73418808-vmss000000  Created container ingress-azure
  Normal   Started    21m                  kubelet, aks-default-73418808-vmss000000  Started container ingress-azure
  Warning  Unhealthy  58s (x120 over 20m)  kubelet, aks-default-73418808-vmss000000  Readiness probe failed: Get "http://10.1.0.16:8123/health/ready": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

**Ingress Controller Error**

```
ERROR: logging before flag.Parse: I1105 02:53:21.896151       1 utils.go:115] Using verbosity level 3 from environment variable APPGW_VERBOSITY_LEVEL
ERROR: logging before flag.Parse: I1105 02:53:21.896248       1 main.go:78] Unable to load cloud provider config '/etc/appgw/azure.json'. Error: Reading Az Context file "/etc/appgw/azure.json" failed: open /etc/appgw/azure.json: permission denied
```

* Any Azure support tickets associated with this issue.
No

**Comment by akshaysngupta** [comment]
@philthynz can you share the complete logs ? `Unable to load cloud provider config` can be treated as a warning.

**Comment by philthynz** [comment]
I believe I know what went wrong... I ran Terraform over the existing resources again, and it changed a few of the app gateway settings. I imagine that after I install the aad-pod-identity and app gateway containers, it makes some changes to the app gateway resource. Me running Terraform again overwritten these changes and hence the error.

I destroyed th app gateway, containers and rbac assignments and it all came back up as normal.

Lesson learned, don't run Terraform apply again after the pod Identity and gateway containers are deployed.
